Documentation/bpf.rst

```diff
@@ -1181,6 +1181,64 @@ The generated LLVM IR can also be dumped in human readable format through:
 
     $ clang -O2 -Wall -emit-llvm -S -c xdp-example.c -o -
 
+In addition, For conversion debugging information to BTF (BPF Type Format),
+more recent ``llc`` and ``pahole`` are needed and ``llc`` and ``pahole``
+can be probed for BTF support:
+
+::
+
+    $ llc -march=bpf -mattr=help |& grep dwarfris
+      dwarfris - Disable MCAsmInfo DwarfUsesRelocationsAcrossSections.
+      dwarfris - Disable MCAsmInfo DwarfUsesRelocationsAcrossSections.
+      dwarfris - Disable MCAsmInfo DwarfUsesRelocationsAcrossSections.
+    $ pahole --help | greo BTF
+    -J, --btf_encode           Encode as BTF
+
+Recent pahole version can be gotten from https://git.kernel.org/pub/scm/devel/pahole/pahole.git
+and note that the ``llvm-objcopy`` tool is required for ``pahole``, so probe it, too.
+
+``-mattr=dwarfris`` should be used with llc and enables ``pahole`` to dump structures correctly.
+(For ``pahole``, dwarf cross-section relation should be disabled through ``-mattr=dwarfris``,
+because BPF backend is not yet implemented in elfutils backend)
+
+Like the above two steps, generating a binary LLVM IR intermediate file
+``xdp-example.bc`` with debugging informantion by adding ``-g``, which
+can later on be passed to llc with ``-mattr=dwarfris``:
+
+::
+
+    $ clang -O2 -g -Wall -target bpf -emit-llvm -c xdp-example.c -o xdp-example.bc
+    $ llc xdp-example.bc -march=bpf -mattr=dwarfris -filetype=obj -o xdp-example.o
+
+And then ``pahole`` can properly dump structures of the BPF program:
+
+::
+
+    $ pahole xdp-example.o
+    struct xdp_md {
+            __u32                      data;                 /*     0     4 */
+            __u32                      data_end;             /*     4     4 */
+            __u32                      data_meta;            /*     8     4 */
+
+            /* size: 12, cachelines: 1, members: 3 */
+            /* last cacheline: 12 bytes */
+    };
+
+Through the option ``-J``, ``pahole`` can convert dwarf format to BTF:
+
+::
+
+    $ pahole -J xdp-example.o
+
```

Comment:
Could you also describe here whether DWARF is discarded from the obj file or retained and only BTF additionally added. I think it's the latter, but would be good to clarify here.

Reply:
Ah, I rechecked the information from `readelf -a xdp-example.o` after `pahole -J xdp-example.o`,
but DWARF still remains like below:
So will describe this, right ?
```c
$ clang -target bpf -O2 -g -c -Xclang -target-feature -Xclang +dwarfris xdp-example.c
$ pahole -J xdp-example.o
$ readelf -a xdp-example.o
[...]
  [ 4] .debug_str        PROGBITS         0000000000000000  00000054
       0000000000000180  0000000000000001  MS       0     0     1
  [ 5] .debug_abbrev     PROGBITS         0000000000000000  000001d4
       00000000000000bc  0000000000000000           0     0     1
  [ 6] .debug_info       PROGBITS         0000000000000000  00000290
       00000000000000fc  0000000000000000           0     0     1
  [ 7] .debug_ranges     PROGBITS         0000000000000000  0000038c
       0000000000000000  0000000000000000           0     0     1
  [ 8] .debug_macinfo    PROGBITS         0000000000000000  0000038c
       0000000000000001  0000000000000000           0     0     1
  [ 9] .debug_pubnames   PROGBITS         0000000000000000  0000038d
       000000000000002d  0000000000000000           0     0     1
  [10] .debug_pubtypes   PROGBITS         0000000000000000  000003ba
       0000000000000058  0000000000000000           0     0     1
  [11] .debug_frame      PROGBITS         0000000000000000  00000418
       0000000000000028  0000000000000000           0     0     8
  [12] .debug_line       PROGBITS         0000000000000000  00000440
       0000000000000082  0000000000000000           0     0     1
  [13] .symtab           SYMTAB           0000000000000000  000004c8
       0000000000000060  0000000000000018          18     2     8
  [14] .rel.debug_info   REL              0000000000000000  00000528
       0000000000000030  0000000000000010          13     6     8
  [15] .rel.debug_frame  REL              0000000000000000  00000558
       0000000000000010  0000000000000010          13    11     8
  [16] .rel.debug_line   REL              0000000000000000  00000568
       0000000000000010  0000000000000010          13    12     8
  [17] .llvm_addrsig     LOOS+0xfff4c03   0000000000000000  00000578
       0000000000000002  0000000000000000   E       0     0     1
  [18] .strtab           STRTAB           0000000000000000  0000057a
       00000000000000f7  0000000000000000           0     0     1
  [19] .BTF              PROGBITS         0000000000000000  00000671
       0000000000000190  0000000000000000           0     0     1
[...]
```